src/osemgrep/networking/Semgrep_App.ml

```diff
@@ -455,4 +497,17 @@ let upload_rule_to_registry caps json =
     ]
   in
   let body = JSON.string_of_json (JSON.from_yojson json) in
-  Http_helpers.post ~body ~headers caps#network url
+  match%lwt Http_helpers.post ~body ~headers caps#network url with
+  | Ok { body = Ok body; _ } -> Lwt.return_ok body
+  | Ok { body = Error msg; code; _ } ->
```

Comment:
the error management seems a bit heavier now ... it's not possible to factorize
a bit so we just handle Ok body = Ok body, and then whatever else as a general error?